docs/source/use_with_tensorflow.mdx

```diff
@@ -0,0 +1,195 @@
+>>> ds = Dataset.from_dict({"data": data}, features=features) 
+>>> ds = ds.with_format("tf")  
+>>> ds[:3]
+{'data': <tf.Tensor: shape=(3,), dtype=int64, numpy=array([0, 0, 1])>
+```
+
+Strings are also supported:
+
+```py
+>>> from datasets import Dataset, Features 
+>>> text = ["foo", "bar"]
+>>> data = [0, 1] 
+>>> ds = Dataset.from_dict({"text": text, "data": data})  
+>>> ds = ds.with_format("tf") 
+>>> ds[:2]
+{'text': <tf.Tensor: shape=(2,), dtype=string, numpy=array([b'foo', b'bar'], dtype=object)>,
+ 'data': <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1])>}
+```
+
+But if you want, you can explicitly format certain columns and leave the other columns unformatted:
+
+```py
+>>> ds = ds.with_format("tf", columns=["data"], output_all_columns=True)
+>>> ds[:2]
+{'data': <tf.Tensor: shape=(2,), dtype=int64, numpy=array([0, 1])>,
+ 'text': ['foo', 'bar']}
+```
+
+The [`Image`] and [`Audio`] feature types are not supported yet.
+
+## Data loading
+
+### Interoperability with tf.data
+
+Although you can load individual samples and batches just by indexing into your dataset, this won't work if you want
+to use Keras methods like `fit()` and `predict()`. You could write a generator function that shuffles and loads batches
+from your dataset and `fit()` on that, but that sounds like a lot of unnecessary work. Instead, if you want to stream
+data from your dataset on-the-fly, we recommend converting your dataset to a `tf.data.Dataset` using the
+`to_tf_dataset()` method.
+
+The `tf.data.Dataset` class covers a wide range of use-cases - it is often created from Tensors in memory, or using a load function to read files on disc
+or external storage. The dataset can be transformed arbitrarily with the `map()` method, or methods like `batch()`
+and `shuffle()` can be used to create a dataset that's ready for training. These methods do not modify the stored data
+in any way - instead, the methods build a data pipeline graph that will be executed when the dataset is iterated over,
+usually during model training or inference. This is different from the `map()` method of Hugging Face `Dataset` objects,
+which runs the map function immediately and saves the new or changed columns.
+
+Since the entire data preprocessing pipeline can be compiled in a `tf.data.Dataset`, this approach allows for massively
+parallel, asynchronous data loading and training. However, the requirement for graph compilation can be a limitation,
+particularly for HuggingFace tokenizers, which are usually not (yet!) compilable as part of a TF graph. As a result, 
+we usually advise pre-processing the dataset as a Hugging Face dataset, where arbitrary Python functions can be
+used, and then converting to `tf.data.Dataset` afterwards using `to_tf_dataset()` to get a batched dataset ready for
+training. To see examples of this approach, please see the example scripts(link) or notebooks(link) for `transformers`.
```

Comment:
```suggestion
training. To see examples of this approach, please see the [example scripts](https://github.com/huggingface/transformers/tree/main/examples/tensorflow) or [notebooks](https://github.com/huggingface/transformers/tree/main/notebooks#tensorflow-examples) for `transformers`.
```

Reply:
I pushed this in my own branch, but it's fixed!